### ISSUE TITLE
Implement Checks to Block Merging of PRs with Status-go Version Pointing to a Branch

### ISSUE THREAD

**Issue by ajayesivan** [status-im/status-react#17802]
### Problem
In our workflow, when making changes related to `status-go`, it's customary to modify the `status-go-version.json` file, pointing it to a branch rather than a release tag. However, it is easy to overlook this step, leading to the unintentional merging of PRs without pushing a new tag and updating the `status-go` version to point to that tag, causing inconsistencies and issues in our codebase.

### Solution
We can implement a new safeguard in our workflow. When a PR modifies the `status-go-version.json` file and points to a branch, it will be automatically blocked from merging. This safeguard will remind contributors to create a release tag and update the status-go version to the new tag before proceeding with the merge.

**Comment by yqrashawn** [comment]
Talked to @jakubgs about this. Seems there's an ongoing discussion about improving the way we link status-go to status-mobile, and @cammellos wanted to mark status-go breaking changes with commit message.

Unassign myself. We might don't need this if there's a better solution.

**Comment by jakubgs** [comment]
This is correct. @cammellos is working on improving this workflow. The ultimate goal is to use versions correctly instead of tagging every other commit and instead to use `git describe`.
